transport/conn.go

```diff
@@ -171,7 +183,7 @@ func (c *connReader) handler(streamID frame.StreamID) ResponseHandler {
 	return h
 }
 
-func (c *connReader) loop() {
+func (c *connReader) loop(ctx context.Context) {
 	c.bufw = frame.BufferWriter(&c.buf)
 	for {
 		resp := c.recv()
```

Comment:
ctx is unused and looking like this loop blocks on recv, which should be interruptable

Reply:
Thanks for noticing. Actually reader should not take the context as this recv can only be interrupted by its underlying connection being closed, which is already done in cluster/pool when it sees that the context is done.